universal-application-tool-0.0.1/app/services/question/TextQuestionDefinition.java

```diff
@@ -0,0 +1,27 @@
+package services.question;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+import java.util.Locale;
+import java.util.Optional;
+
+public class TextQuestionDefinition extends QuestionDefinition {
+
+  @JsonCreator
+  public TextQuestionDefinition(
+      @JsonProperty("id") long id,
+      @JsonProperty("version") long version,
+      @JsonProperty("name") String name,
+      @JsonProperty("path") String path,
+      @JsonProperty("description") String description,
+      @JsonProperty("questionText") ImmutableMap<Locale, String> questionText,
+      @JsonProperty("questionHelpText") Optional<ImmutableMap<Locale, String>> questionHelpText) {
+    super(id, version, name, path, description, questionText, questionHelpText);
+  }
+
+  @Override
+  public QuestionType getQuestionType() {
+    return QuestionType.TEXT;
+  }
+}
```

Comment:
This should also implement the following method:

```
 /** Get a map of scalars stored by this question definition. */
  @JsonIgnore
  public ImmutableMap<String, ScalarType> getScalars() {
    return ImmutableMap.of("text", ScalarType.STRING);
  }
```

...but if I recall we were going to make QuestionDefinition abstract in another PR, so you can do it then.